Source/Core/Core/IOS/ES/Views.cpp

```diff
@@ -260,6 +260,39 @@ IPCCommandResult ES::DIGetTicketView(const IOCtlVRequest& request)
   return GetDefaultReply(IPC_SUCCESS);
 }
 
+IPCCommandResult ES::DIGetTMDSize(const IOCtlVRequest& request)
+{
+  if (!request.HasNumberOfValidVectors(0, 1) || request.io_vectors[0].size != sizeof(u32))
+    return GetDefaultReply(ES_EINVAL);
+
+  if (!GetTitleContext().active)
+    return GetDefaultReply(ES_EINVAL);
+
+  Memory::Write_U32(static_cast<u32>(GetTitleContext().tmd.GetRawTMD().size()),
+                    request.io_vectors[0].address);
+  return GetDefaultReply(IPC_SUCCESS);
+}
+
+IPCCommandResult ES::DIGetTMD(const IOCtlVRequest& request)
+{
+  if (!request.HasNumberOfValidVectors(1, 1) || request.in_vectors[0].size != sizeof(u32))
+    return GetDefaultReply(ES_EINVAL);
+
+  const u32 tmd_size = Memory::Read_U32(request.in_vectors[0].address);
+  if (tmd_size != request.io_vectors[0].size)
+    return GetDefaultReply(ES_EINVAL);
+
+  if (!GetTitleContext().active)
```

Comment:
It could, but I put the checks in the same order as they appear in IOS: ioctlv handler checks vectors, then calls an internal function `ES_DiGetTmd(TMD* tmd, u32* size)` and it is that function which checks for the title context.

Tbh, it's unlikely to make any difference :P